src/com/senlainc/warsaw/tyurin/service/CraftsmanService.java

```diff
@@ -36,19 +35,28 @@ public void addCraftsman(Craftsman craftsman) {
     }
 
     @Override
-    public void removeCraftsmen(Craftsman craftsman) {
-        craftsmanDAO
-                .getCraftsmen()
-                .remove(craftsman);
+    public void removeCraftsmanById(long id) {
+
+        List<Craftsman> craftsmen = craftsmanDAO.getCraftsmen();
+
+        for (Craftsman craftsman : craftsmen) {
```

Comment:
Use stream API

Reply:
The idea was that I break the loop when found craftsman I need.
Some unneeded iterations could happen if I use stream API

Reply:
The stream API has vertical execution of operations (except sorting). This means that once the element passes the filter, find first will return a value and end the stream. Additional operations are not required

https://akifhatipoglu.medium.com/guide-to-java-stream-api-stream-execution-order-4c2df14d1d00

Reply:
Now it is clear to me, thanks!